### PR TITLE
`Communication`: Update plagiarism case posts automatically with new replies

### PR DESCRIPTION
**Pull request by nityanandaz**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Replies in plagiarism cases and do not update correctly while the chat is open.

### Description

The client handling for new messages has been updated to respect all messages for all types of conversations and plagiarism cases correctly.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Text Exercise 

Plagiarism Cases:
1. Hand in the same answer with the 2 students for the exercise
2. Generate a plagiarism case using the instructor
3. Select one of the students and notify them
4. Use the selected student to reply to the plagiarism case, the answers should update for the instructor

Communication tab:
1. Open the communication tab with a user
2. Write messages in any channel with another user, only new messages in course-wide channels should be automatically added at the top
3. Repeat with a set of selected channels using the filter option. Only messages from the selected channels should be automatically added at the top

Messages tab:
1. Open a conversation with a user
2. Write messages in this and other conversations with another use. Only messages from the conversation should be shown
3. Repeat 1-2 for each type of conversation: course-wide channel, private/public channel, group chat, one-to-one chat

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

No UI Changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced filtering in the `MetisService` to better handle posts related to plagiarism cases.
	- Improved conditions for displaying posts from selected course-wide channels.

- **Tests**
	- Added tests for new filtering logic and for handling messages in private channels and plagiarism cases.

### CODE REVIEW

File: src/test/javascript/spec/helpers/sample/metis-sample-data.ts [hunk 1]
```diff
@@ -283,6 +284,15 @@ export const metisPostInChannel = {
...
 } as Post;
 
+export const plagiarismPost = {
+    id: 11,
+    author: metisUser1,
+    title: 'title',
+    content: 'plagiarism Case',
+    creationDate: undefined,
+    plagiarismCase: { id: 1 } as PlagiarismCase,
+} as Post;
```
[the remark this answers is not in the document]
Comment (lennart-keller) [reply src/test/javascript/spec/helpers/sample/metis-sample-data.ts]: The creation date is not needed.

**Review by MaximilianAnzinger — DISMISSED** [review]
Code LGTM

**Review by konrad2002 — APPROVED** [review]
manually tested on legacy ts1, works as expected

**Review by rstief — DISMISSED** [review]
Code looks good 👍

**Review by basak-akan — APPROVED** [review]
Tested on TS2, worked as expected

**Review by georgyia — DISMISSED** [review]
manually tested on ts2, works as expected

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by basak-akan — APPROVED** [review]
re-approve

**Review by rstief — APPROVED** [review]
reapproved

**Review by reschandreas — APPROVED** [review]
tested locally, works as expected

**Review by georgyia — APPROVED** [review]
reapprove

**Pull request merged**